File: packages/java-sdk/src/main/java/com/turbodocx/models/EmbeddedSignatureRecipientResult.java
package com.turbodocx.models;

/**
 * One resolved signer in a {@link CreateEmbeddedSignatureResponse}.
 */
public class EmbeddedSignatureRecipientResult {
    private final String recipientId;
    private final String name;
    private final String email;
    /**
     * The embeddable signing URL — present only when it is this recipient's turn
     * ({@code status == "ready"}). Null for a recipient who cannot sign yet ({@code "pending"}) or
     * has already signed ({@code "completed"}); mint it later with
     * {@code TurboSign.createSigningUrl}.
     */
    private final String embedUrl;
    /** One of {@code "ready"}, {@code "pending"}, {@code "completed"}. */
    private final String status;
    /** One of {@code "otp"}, {@code "external_idv"}, {@code "override"}, or null. */
    private final String identityVerificationMode;

    public EmbeddedSignatureRecipientResult(String recipientId, String name, String email, String embedUrl,
                                            String status, String identityVerificationMode) {
        this.recipientId = recipientId;
        this.name = name;
        this.email = email;
        this.embedUrl = embedUrl;
        this.status = status;
        this.identityVerificationMode = identityVerificationMode;
    }

    public String getRecipientId() {
        return recipientId;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public String getEmbedUrl() {
        return embedUrl;
    }

    public String getStatus() {
        return status;
    }

    public String getIdentityVerificationMode() {
        return identityVerificationMode;
    }
}

File: packages/php-sdk/src/Types/Responses/EmbeddedSignatureRecipientResult.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Responses;

/**
 * One resolved signer in a {@see CreateEmbeddedSignatureResponse}. Mirrors the TypeScript SDK's
 * `EmbeddedSignatureRecipientResult`.
 */
final class EmbeddedSignatureRecipientResult
{
    public const STATUS_READY = 'ready';
    public const STATUS_PENDING = 'pending';
    public const STATUS_COMPLETED = 'completed';

    /**
     * @param string $recipientId TurboDocx recipient id
     * @param string $name Recipient's full name
     * @param string $email Recipient's email address
     * @param string|null $embedUrl The embeddable signing URL — present only when it is this
     *     recipient's turn (`status: 'ready'`). Null for 'pending' or 'completed'.
     * @param string $status 'ready' | 'pending' | 'completed'
     * @param string|null $identityVerificationMode 'otp' | 'external_idv' | 'override' | null
     */
    public function __construct(
        public string $recipientId,
        public string $name,
        public string $email,
        public ?string $embedUrl,
        public string $status,
        public ?string $identityVerificationMode,
    ) {}
}
